src/lib/events/events.ts

```diff
@@ -10,14 +10,16 @@ export class Events {
     constructor(private readonly http: AxiosInstance, private readonly config: TimelyAppConfig) {}
 
     async getAll(start?: string, end?: string): Promise<TimelyEvent[]> {
```

Comment:
For consumers of the library, it may be more idiomatic if we allowed the passing of a `Date` type instead of an ISO string, or kept things as-is but provided a type guard to validate that it is a proper ISO string - which do you think would be preferable?

Reply:
@CalebAlbers I think the most flexible approach would be to accept a `Date` *or* a `string`. Something like this:

```
export type DateString = string | Date

ensureISOFormat(date: DateString): string {
    try {
        const re = /^\d{4}-\d{2}-\d{2}$/
        if (typeof date === 'string' && re.test(date)) return date
        return date instanceof Date
            ? this.asShortISOString(date)
            : this.asShortISOString(new Date(date.replace(/-/gi, '/')))
    } catch (e) {
        throw new Error(`Unable to parse ${date} as YYYY-MM-DD`)
    }
}

private asShortISOString(date: Date): string {
    return date.toISOString().slice(0, 10)
}
```
The above seems to work with any funky format I could think of to provide. These tests all pass:

```
describe('ensureISOFormat', () => {
    it('fails given a string that cannot be parsed as a date', () => {
        expect(() => events.ensureISOFormat('i am not a date')).toThrow(
            'Unable to parse i am not a date as YYYY-MM-DD',
        )
    })
    
    it('fails given a type that cannot be parsed as a Date', () => {
        expect(() => events.ensureISOFormat(false as never)).toThrow(
            'Unable to parse false as YYYY-MM-DD',
        )
    })
    
    it('succeeds given mm/dd/yyyy', () => {
        expect(events.ensureISOFormat('01/01/1970')).toEqual('1970-01-01')
    })
    
    it('succeeds given yyyy/mm/dd', () => {
        expect(events.ensureISOFormat('1970/01/01')).toEqual('1970-01-01')
    })
    
    it('succeeds given yyyy-m-d', () => {
        expect(events.ensureISOFormat('1970-1-1')).toEqual('1970-01-01')
    })
    
    it('succeeds given yy-m-d', () => {
        expect(events.ensureISOFormat('70-1-1')).toEqual('1970-01-01')
    })
    
    it('succeeds given m-d-yy', () => {
        expect(events.ensureISOFormat('1-1-70')).toEqual('1970-01-01')
    })
    
    it('succeeds given a Date object', () => {
        expect(events.ensureISOFormat(new Date('1970/01/01'))).toEqual('1970-01-01')
    })
})
```

Thoughts?

Reply:
I like that! Makes it super easy to use imo. Great add